### PR TITLE
feat(bigquery/storage/managedwriter): improve error communication

### PR DESCRIPTION
**Pull request by shollyman**
This PR:
* augments the package docs with information on error handling
* adds an integration test demonstrating use of StorageError details.

Fixes: https://github.com/googleapis/google-cloud-go/issues/6321

### CODE REVIEW

**Review by yirutang — APPROVED** [review]
Cool, thanks!

**Comment by shollyman** [comment]
This needs some refactoring, but I want to get https://github.com/googleapis/gax-go/issues/212 resolved to unblock this work.

**Comment by shollyman** [comment]
Changes landed in gax, backed out the utility method in favor of leveraging the new `ExtractProtoMessage()` added to apierror.

File: bigquery/storage/managedwriter/doc.go [hunk 1]
```diff
@@ -170,5 +170,33 @@ have been finalized, meaning they'll no longer allow further data writes.
...
+
+Specifically, the apierror package (https://pkg.go.dev/github.com/googleapis/gax-go/v2/apierror)
+provides convenience methods for extracting structured information about errors.
+
+Additionally, the BigQuery Storage API will augment applicable errors with additional
+service-specific details in the form of a StorageError message.  Please note that despite the
+name, this protocol buffer message does not implement Go's error interface.  By leveraging
+ExtractProtoMessage, which is provided as part of the apierror packed, you can retrieve
+service-specific details.
+
```
Comment (quartzmo) [inline bigquery/storage/managedwriter/doc.go:187]: The BigQuery Storage API service augments applicable errors with
service-specific details in the form of a StorageError message.  (Note that despite the
name, this lower-level protocol buffer type does not implement Go's error interface.)

The StorageError message is accessed via the (*apierror.ApiError).ExtractProtoMessage
method in the [apierror](https://pkg.go.dev/github.com/googleapis/gax-go/v2/apierror) package:

Comment (shollyman) [reply bigquery/storage/managedwriter/doc.go:188]: Reworked based on your suggestions, PTAL.

**Review by quartzmo — APPROVED** [review]

**Pull request merged**